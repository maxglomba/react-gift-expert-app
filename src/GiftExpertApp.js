import React, { useState } from 'react'
import CategoryAdd from './components/CategoryAdd';
import GifGrid from './components/GifGrid';
import PropTypes from 'prop-types'

const GiftExpertApp = ( {defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>Gift Expert App</h2>
            <hr />
            <CategoryAdd categories={categories} setCategories={setCategories} />

            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}

GiftExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}

export default GiftExpertApp;
