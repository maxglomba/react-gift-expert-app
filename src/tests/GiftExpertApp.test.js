import React from 'react';
import { shallow } from 'enzyme'
import GiftExpertApp from '../GiftExpertApp';

describe('pruebas en componente <GiftExpertApp />', () => { 
    const categories = ['naruto', 'samurai x'];
    const wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);

    test('debe renderizar el componente <GiftExpertApp /> correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    })

    test('debe obtener titulo de aplicaciÃ³n "Gift Expert App"', () => { 
        const contentValue = wrapper.find('h2').text();
        expect(contentValue).toBe('Gift Expert App');
    })

    test('enviarse a <CategoryAdd /> las categorias actuales y la funcion de agregar', () => { 
        expect(wrapper.find('CategoryAdd').prop('categories')).toBe(categories);
        expect(typeof wrapper.find('CategoryAdd').prop('setCategories')).toBe('function');
    })

    test('debe obtener las listas de imagenes de todas las categorias buscadas', () => { 
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

 })