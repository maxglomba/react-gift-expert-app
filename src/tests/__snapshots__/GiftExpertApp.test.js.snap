// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente <GiftExpertApp /> debe renderizar el componente <GiftExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <hr />
  <CategoryAdd
    categories={
      Array [
        "naruto",
        "samurai x",
      ]
    }
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="naruto"
      key="naruto"
    />
    <GifGrid
      category="samurai x"
      key="samurai x"
    />
  </ol>
</Fragment>
`;
